### ISSUE TITLE
Add `ratio-notation`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#6575]
### What is the problem you're trying to solve?

Iv recently discovered that Chrome didn't support decimals in ratio data types, but Firefox did.
e.g. `1.7/1` fails but `10/6` is OK

This is probably due to the different types accepted between Media Queries Level 3 and 4.
ref: https://developer.mozilla.org/en-US/docs/Web/CSS/ratio#syntax

### What solution would you like to see?

- Name: `ratio-notation`
- Primary option: `"number"|"integer"`
- Secondary options: none
- Autofixable: Not initially
- Message: `Expected "${primary}" ratio notation`
- Description: "Specify number or integer notation for ratio data type."
- Section: "Enforce conventions" -> "Notation"

A `foo % 1 === 0` condition should be enough to detect it.

**Comment by Mouvedia** [comment]
>  The *-notation rules are for when the same thing can be represented in two or more ways…

That's the case here.
e.g. `1.5/3` === `3/6`

Depending on the level of Media Queries it's supported or not.
It's akin to the selector-not-notation rule that I added previously.

**Comment by romainmenke** [comment]
This PostCSS plugin fixes the browser issues : https://github.com/csstools/postcss-plugins/tree/postcss-preset-env--v8/plugins/postcss-media-queries-aspect-ratio-number-values#readme

Support stats : https://mrhenry.github.io/web-tests/#de023e11-f61a-4635-a74c-e4a152251bdf

Stylelint could have a rule for it, but since it is perfectly possible to polyfill I think it might make sense to do just that :)

**Comment by Mouvedia** [comment]
@romainmenke I am not using PostCSS, so that wouldn't be a solution _for me_.
It's reassuring to see that I am not the only one who saw the need for a fix, though.

**Comment by FloEdelmann** [comment]
I think that rule should not exist to ensure browser compatibility, but to enforce a convention across the codebase:

---

### Primary option `"number"`

> [!NOTE]
> Can always be autofixed by calculating the division.

The following patterns are considered problems:

```css
a { aspect-ratio: 2 / 1; }
a { aspect-ratio: 3 / 2; }
a { aspect-ratio: 16/9; }
a { aspect-ratio: 1.7/1; }
```

The following patterns are *not* considered problems:

```css
a { aspect-ratio: 2; }
a { aspect-ratio: 1.5; }
a { aspect-ratio: 1.7778; }
a { aspect-ratio: 1.7; }
```

### Primary option `"integers"`

> [!NOTE]
> If the value is a single integer, it can be autofixed by appending `/1`.
> For decimals, it could use a [more sophisticated algorithm](https://stackoverflow.com/questions/5124743/algorithm-for-simplifying-decimal-to-fractions) for finding a human-readable fraction that approximates the decimal.

The following patterns are considered problems:

```css
a { aspect-ratio: 2; }
a { aspect-ratio: 1.5; }
a { aspect-ratio: 1.7778; }
a { aspect-ratio: 1.7/1; }
```

The following patterns are *not* considered problems:

```css
a { aspect-ratio: 2 / 1; }
a { aspect-ratio: 3 / 2; }
a { aspect-ratio: 16/9; }
a { aspect-ratio: 17/10; }
```

#### Secondary option `ignoreDecimalFraction: true` (default: `false`)

The following patterns are considered problems:

```css
a { aspect-ratio: 2; }
a { aspect-ratio: 1.5; }
a { aspect-ratio: 1.7778; }
```

The following patterns are *not* considered problems:

```css
a { aspect-ratio: 2 / 1; }
a { aspect-ratio: 3 / 2; }
a { aspect-ratio: 16/9; }
a { aspect-ratio: 1.7/1; }
```

**Comment by romainmenke** [comment]
Hi @FloEdelmann,

Thank you for the overview.

It might make sense to follow the wording of the specification? https://www.w3.org/TR/css-values-4/#ratios

- `2 / 3` -> `ratio`
- `1.333` -> `number`

------

Is there a specific use case for `ignoreDecimalNumerator`?
To me it seems that this mainly exists in you proposal to cover all possible options, but I might be overlooking a genuine case where you would want to enforce `ratio` while allowing fractions in the number components.

**Comment by Mouvedia** [comment]
@romainmenke I prefer following what I have referenced, which is https://developer.mozilla.org/en-US/docs/Web/CSS/ratio#syntax.
i.e. `integer` / `number`
see https://drafts.csswg.org/css-syntax-3/#consume-number

**Comment by romainmenke** [comment]
There are indeed three types :
- `integer`
- `number`
- `ratio`

With my first comment I only wanted to address using the `fraction` label for `2 / 3` where the specification uses `ratio`.

**Comment by Mouvedia** [comment]
tl;dr

what I am proposing:

```
aspect-ratio: <integer>/<integer>; // level 3
// vs
aspect-ratio: <number>/<number>; // level 4
```

The rule he is proposing which should have its own separate issue:

```
aspect-ratio: <number>; // "number"
// vs
aspect-ratio: <number>/<number>; // "fraction"
```

**Comment by Mouvedia** [comment]
Thinking about it more, this would work too:

- Name: ratio-notation
- Primary option: "number"|"fraction"
- Secondary options: "integer-only"
- Autofixable: true
- Message: Expected "${primary}" ratio notation / Expected "${primary}" to be an integer
- Description: "Specify number or fraction notation for ratio data type."
- Section: "Enforce conventions" -> "Notation"
